proteinbenchmark/system_setup.py

```diff
@@ -618,38 +669,96 @@ def solvate(
             hydrogenMass=hydrogen_mass,
             switchDistance=switch_distance,
         )
+    elif sim_platform == 'gmx' and smirnoff == False:
+        switch_distance = nonbonded_cutoff - vdw_switch_width
+        openmm_system = force_field.createSystem(
+            modeller.topology,
+            nonbondedMethod=app.PME,
+            nonbondedCutoff=nonbonded_cutoff,
+            rigidWater=False,
+            hydrogenMass=hydrogen_mass,
+            switchDistance=switch_distance,
+        )
+    
+    if (smirnoff and sim_platform != 'gmx') or smirnoff == False:
+        # Validate total charge of solvated system
+        for i in range(openmm_system.getNumForces()):
+            if isinstance(openmm_system.getForce(i), openmm.NonbondedForce):
+                nonbonded_force = openmm_system.getForce(i)
+                break
 
-    # Validate total charge of solvated system
-    for i in range(openmm_system.getNumForces()):
-        if isinstance(openmm_system.getForce(i), openmm.NonbondedForce):
-            nonbonded_force = openmm_system.getForce(i)
-            break
-
-    else:
-        raise ValueError("The ForceField does not specify a NonbondedForce")
-
-    solvated_total_charge = unit.Quantity(0, unit.elementary_charge)
-    for i in range(nonbonded_force.getNumParticles()):
-        solvated_total_charge += nonbonded_force.getParticleParameters(i)[0]
+        else:
+            raise ValueError("The ForceField does not specify a NonbondedForce")
 
-    solvated_total_charge = int(
-        numpy.round(solvated_total_charge.value_in_unit(unit.elementary_charge))
-    )
+        solvated_total_charge = unit.Quantity(0, unit.elementary_charge)
+        for i in range(nonbonded_force.getNumParticles()):
+            solvated_total_charge += nonbonded_force.getParticleParameters(i)[0]
 
-    if solvated_total_charge != 0:
-        raise ValueError(
-            f"Total charge of solvated system is {solvated_total_charge:d}"
+        solvated_total_charge = int(
+            numpy.round(solvated_total_charge.value_in_unit(unit.elementary_charge))
         )
 
-    # Write OpenMM system to XML file
-    write_xml(openmm_system_xml, openmm_system)
-
+        if solvated_total_charge != 0:
+            raise ValueError(
+                f"Total charge of solvated system is {solvated_total_charge:d}"
+                )
+    
+    if sim_platform != 'gmx':
+        # Write OpenMM system to XML file
+        write_xml(openmm_system_xml, openmm_system)
+    else:
+        import parmed as pmd
+        if smirnoff:
+            interchange = force_field.createInter(modeller.topology, force_field)
+            interchange.positions = modeller.positions
+            
+            openmm_sys = interchange.to_openmm()
```

Comment:
Interchange should be able to write directly to gro and top files without going through parmed.